### PR TITLE
Modify rule S2323: LaYC format

### PR DESCRIPTION
**Pull request by marco-antognini-sonarsource**
I felt the "how to fix it" section was unnecessary given how trivial this is. Let me know if you disagree.

## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [x] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S2323/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
+
+* {cpp} reference - https://en.cppreference.com/w/cpp/language/translation_phases[Phases of translation]
+
+=== External coding guidelines
+
 * MISRA C:2012, 3.2 - Line-splicing shall not be used in // comments
 
+=== Related rules
+
+* S1916 is a companion rule and detects trailing whitespace after `\`.
```
Comment (amelie-renard-sonarsource) [inline rules/S2323/cfamily/rule.adoc:52]:
```suggestion
* S1916 is a companion rule and detects trailing whitespace after `\`
```

File: rules/S2323/cfamily/rule.adoc [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
 
-Line-splicing occurs when the \ character is immediately followed by a new-line character. If the source line containing a ``++//++`` comment ends with a '\', the next line becomes part of the comment. This may result in unintentional removal of code.
+When a line of code ends with the `\` (backslash) character, the newline character is deleted.
+The compiler considers the next line a continuation of the current line, resulting in only one logical line of code.
 
+This source code transformation, known as _line-splicing_, occurs during the second phase of C and {cpp} translation.
 
-=== Noncompliant code example
+_Line-splicing_ does not apply inside {cpp}11 raw string literal.
+Furthermore, it is often harmless and used to improve the readability of macros:
```
Comment (amelie-renard-sonarsource) [inline rules/S2323/cfamily/rule.adoc:9]: The information in these sentences is relevant, but I find the logic hard to follow. What about:
```suggestion
_Line-splicing_ is often harmless and is sometimes used to improve the readability of macros:
[source,c]
----
#define LOG_FAILURE(CONDITION, MESSAGE) 
  do {                                  \
    if (!(CONDITION)) {                 \
      log(MESSAGE);                     \
    }                                   \
  } while (0)
----

It is interesting to note that _Line-splicing_ does not apply inside {cpp}11 raw string literal.

```

**Review by amelie-renard-sonarsource — CHANGES_REQUESTED** [review]
Nice work! I like the information and the example you added
I have one suggestion

File: rules/S2323/cfamily/rule.adoc (hunk 1 above)
Comment (marco-antognini-sonarsource) [reply rules/S2323/cfamily/rule.adoc:52]: The full stop is required since it's a full sentence.

File: rules/S2323/cfamily/rule.adoc (hunk 2 above)
Comment (marco-antognini-sonarsource) [reply rules/S2323/cfamily/rule.adoc:9]: Makes sense, thanks for the suggestion. I'll amend the description.
